.github/workflows/deploy.yaml

```diff
@@ -56,11 +56,66 @@ jobs:
           yarn util:link
           yarn build
 
+  publish_npm:
+    name: Publish to NPM
+    runs-on: ubuntu-latest
+    # we must wait for the possible commit changes from the smart contract
+    # deployment for a full version
+    needs: [build, deploy_smart_contracts]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # we use a custom PAT to allow our workflows to run on this commit,
+          # which is not the case normally
+          token: ${{ secrets.COMMIT_BOT_ACCESS_TOKEN }}
+
+      - name: Configure Git info
+        run: ./scripts/configure-git-info.sh
+
+      - name: Use Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+          registry-url: https://registry.npmjs.org/
+
+      - name: Install Yarn (only when using ACT)
+        if: env.ACT != ''
+        run: |
+          curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+          echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+          apt update
+          apt install --no-install-recommends yarn
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn config get cacheFolder)"
+
+      - name: Restore cache of node modules and yarn cache
+        uses: actions/cache@v2
+        id: nodejs-cache
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+            ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-nodejs-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-nodejs-
+
+      - name: Install dependencies / link / build
+        run: |
+          yarn
+          yarn setup
+          yarn util:link
+          yarn build
+
       - name: Publish prerelease version (On pushes to master)
         if: github.ref == 'refs/heads/master'
         run: |
-          # skip releasing for changes from our bots to prevent circular change propagation
-          [ "noreply@hoprnet.org" = "$(git show -s --format='%ae' ${HOPR_GITHUB_REF})" ] && exit 0
+          # skip releasing for changes which we committed here before to
```

Comment:
isn't it the version bot that will trigger the versioning commit that should get published?

Reply:
The version bot is the service account which these actions are running under. So yes, that bot will end up doing the commit.

Reply:
obsolete